pkgs/servers/sql/postgresql/default.nix

```diff
@@ -6,6 +6,9 @@ let
       , glibc, zlib, readline, openssl, icu, systemd, libossp_uuid
       , pkgconfig, libxml2, tzdata
 
+      # This is important to obtain a version of `libpq` that does not depend on systemd.
+      , enableSystemd ? (lib.versionAtLeast "9.6" version && !stdenv.isDarwin)
```

Comment:
maybe also add an assert that systemd can't be null when `enableSystemd = true`. Though this is minor